src/pudl_rmi/validate.py

```diff
@@ -0,0 +1,178 @@
+        npb.convert_dtypes(convert_floating=False)
+        .rename(columns=RMI_TO_PUDL_COLS)
+        .replace({"ferc_acct_name": RMI_TO_PUDL_FERC_ACCT_NAME})
+        .assign(report_date=lambda x: pd.to_datetime(x.report_year, format="%Y"))
+        .merge(  # we need the pudl id!!
+            utils_f1[["utility_id_ferc1", "utility_id_pudl"]],
+            on=["utility_id_ferc1"],
+            how="left",
+            validate="m:1",
+        )
+    )
+    npb.loc[:, "ferc_acct_name"] = npb.ferc_acct_name.str.title()
+    return npb
+
+
+def add_data_source(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Add a data_source column with 'FERC' and 'PUC'.
+
+    The depreciation data includes a `data_source` column as a part of the
+    primary key of that data set. This is because there are sometimes we have
+    depreciation study from both `FERC` and `PUC` for the same utility and
+    plants. Because of this, in order to compare the FERC plant input tables to
+    the FERC-EIA-Depreciation tables we need to add *both* versions of the
+    depreciation `data_source`.
+    """
+    if "data_source" not in df:
+        df = pd.concat(
+            [
+                df.assign(data_source="PUC"),
+                df.assign(data_source="FERC"),
+            ]
+        )
+    return df
+
+
+def group_sum_cols(df, data_cols: List[str], by: List[str]) -> pd.DataFrame:
+    """Groupby sum a specific table's data cols."""
+    # convert date to year bc many of the og depish studies are EOY
+    summed_out = (
+        df.assign(report_year=lambda x: x.report_date.dt.year)
+        .astype({"report_year": pd.Int64Dtype()})
+        .groupby(by=by, dropna=True)[data_cols]
+        .sum(min_count=1)
+    )
+    return summed_out
+
+
+def agg_test_data(
+    df1: pd.DataFrame, df2: pd.DataFrame, data_cols: List[str], by: List[str], **kwarg
+) -> pd.DataFrame:
+    """
+    Merge two grouped input tables to determine if summed data column are equal.
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``.
+        df2: Other dataframe to sum and check consistency against ``df1``.
+        data_cols: data columns to check. Columns must be in both ``df1`` and
+            ``df2``.
+    """
+    test = pd.merge(
+        group_sum_cols(df1.pipe(add_data_source), data_cols=data_cols, by=by),
+        group_sum_cols(df2.pipe(add_data_source), data_cols=data_cols, by=by),
+        right_index=True,
+        left_index=True,
+        suffixes=("_1", "_2"),
+        how="outer",
+    ).astype("float64")
+    for data_col in data_cols:
+        test.loc[:, f"{data_col}_isclose"] = np.isclose(
+            test[f"{data_col}_1"],
+            test[f"{data_col}_2"],
+            equal_nan=True,
+            **kwarg,
+        )
+        test.loc[:, f"{data_col}_ratio"] = test[f"{data_col}_1"] / test[f"{data_col}_2"]
+        logger.info(
+            f"{data_col} close records: {len(test[test[f'{data_col}_isclose']])/len(test):.02%}"
```

Comment:
IIRC you can also use `boolean_col.sum()` to count the number of true values, rather than needing to get the length of a masked column, which can be hard to read. Also the doubly nested f-strings are kind of an eyefull. Maybe...
```py
isclose_fraction = test[f"{data_col}_isclose"].sum() / len(test)
logger.info(f"{data_col} close records: {isclose_fraction}:.02%}")
```